src/Compilers/Core/Portable/MetadataReference/ModuleMetadata.cs

```diff
@@ -22,58 +22,58 @@ public sealed partial class ModuleMetadata : Metadata
         private readonly PEModule _module;
 
         /// <summary>
-        /// Optional data that should be kept alive as long as this <see cref="ModuleMetadata"/> is alive.  This can be
-        /// useful, for example, if there is backing memory that the metadata depends on that should be kept rooted so it
-        /// doesn't get garbage collected.
+        /// Optional action to invoke when this metadata is disposed.
         /// </summary>
-        private readonly IDisposable? _owner;
+        private readonly Action? _onDispose;
 
         /// <summary>
-        /// Whether or not <see cref="_owner"/> should be <see cref="IDisposable.Dispose"/>'d when this object is
-        /// Disposed.  Is controlled by the <c>leaveOpen</c> flag in <see cref="CreateFromStream(Stream, bool)"/>, or
-        /// the <see cref="PEStreamOptions.LeaveOpen"/> flag in <see cref="CreateFromStream(Stream, PEStreamOptions)"/>.
+        /// Whether or not to call _onDispose when this is disposed.  The reason that this is controlled by a flag is
```

Comment:
>Whether or not to call _onDispose when this is disposed. [](http://example.com/codeflow?start=12&length=56)

Could we clear `_onDispose` when copying instead of adding a separate field?

Reply:
Done!